lib/cc/engine/analyzers/ruby/main.rb

```diff
@@ -24,6 +24,10 @@ class Main < CC::Engine::Analyzers::Base
 
           def process_file(file)
             RubyParser.new.process(File.binread(file), file, TIMEOUT)
+          rescue Timeout::Error
+            warn "TIMEOUT parsing #{file}. Skipping."
+          rescue => e
```

Comment:
> Wouldn't this make more sense?

That's actually pretty similar what the code used to be! But there's a good reason we don't do it that way any more.

The reason we don't use `case ... when *RESCUABLE_ERRORS` (or `rescue *RESCUABLE_ERRORS =>`, which has very similar semantics), is that when that collection includes class `A`, and the raised exception is of class `B < A`, it matches. Which seems sensible at first brush.

Unfortunately, you'll note that we rescue `::RuntimeError`. We do this because there are some bare `raise`s we hit on parser issues (I think somewhere in `RubyParser`, or possibly even in `Racc`?) that we wanted to skip, but we felt rescuing from all potential children of `RuntimeError` was excessive. So we have the logic you see today which only matches exact instances of rescuable exception classes (though even the current logic for that could probably be cleaned up without changing the behavior).

> It was a RegexpError and not in the list
> I don't actually like AnalyzerBase::RESCUABLE_ERRORS...That's breaking responsibilities and should be pushed down

I agree with you, and I'd be happy to see a PR break that down so that list is per-analyzer-class, but since this PR is focused on MRI, maybe that should be a separate PR? Same goes for the timeout handling, actually: AFAICT we weren't rescuing those before on JRuby, so that also would make sense to me as a separate changeset.

So maybe:

1. Back out the two rescues here so this is more focused on getting us on MRI
2. Open another PR to:
    - Rescue timeouts globally (since I believe all analyzers do raise them, correct me if I missed something)
   - Break out rescuable exception classes per-analyzer-class
   - ... and add `RegexpError` to the errors the ruby parser will catch.

Would that make sense?